chapters/Update-Beacons.md

```diff
@@ -243,37 +280,63 @@ signing the PSBT, a valid, spendable Bitcoin transaction can be created by
-   Bitcoin transaction is not a ::Beacon Signal::, return.
-1. Set `cid` to the 32 byte ::CID:: in `txOut`.
-1. Initialize a `didUpdateBundle` variable to the result of retrieving the `cid`
-   from the ::CAS::. If unable to retrieve `cid`, MUST throw a `latePublishError` and
-   treat the DID being resolved as invalid.
-1. Set `didUpdateCID` to the value of `didUpdateBundle[document.id]`.
-1. If `didUpdateCID` is empty return null (this ::Beacon Signal:: contains no update
-   for the DID being resolved).
-1. Set `didUpdatePayload` to the result of retrieving `didUpdateCID` from ::Content
-   Addressable Storage:: (CAS) or ::Sidecar Data::. If `didUpdateCID` not found MUST throw
-   a `latePublishError`.
-1. Return `didUpdatePayload`.
-
-![](https://hackmd.io/_uploads/HkZe90lkp.jpg)
-
-### SMTAggregator Beacon
-
-A SMTAggregator ::Beacon:: is a ::Beacon:: whose ::Beacon Signals:: are Bitcoin transactions
+#### Process CIDAggregate Beacon Signal
+
+A ::Beacon Signal:: from a CIDAggregate Beacon is a Bitcoin transaction that contains
+the `hashBytes` of a ::DID Update Bundle:: in its first transaction output.
+The corresponding ::DID Update Bundle:: MUST either be provided through ::Sidecar Data::
+or by converting `hashBytes` into a IPFS v1 ::Content Identifier:: and attempting to 
+retrieve it from ::Content Addressable Storage::. The ::DID Update Bundle:: maps from 
+**did:btc1** identifiers to hashes of ::DID Update payloads:: applicable for that identifier.
+Again this algorithm attempts to retrieve and validate the ::DID Update Payload:: identified 
+for the identifier being resolved. If successful, the ::DID Update Payload:: is returned.
+
+This algorithm is called by the [Process Beacon Signals] algorithm as part of the
+[Read] operation. It takes as inputs a **did:btc1** identifier, `btc1Identifier`, a 
+::Beacon Signal::, `tx`, and a optional object, `signalSidecarData`, containing any 
+sidecar data provided to the resolver for the ::Beacon Signal:: identified by the 
+Bitcoin transaction identifier.
+
+The algorithm returns the ::DID Update payload:: announced by the ::Beacon Signal::
+for the ::did:btc1:: identifier being resolved or throws an error.
+
+
+1. Initialize a `txOut` variable to the 0th transaction output of the `tx`.
+1. Set `didUpdatePayload` to null.
+1. Check `txOut` is of the format `[OP_RETURN, OP_PUSH32, 32BYTE]`, if not,
+   then return `didUpdatePayload`. The Bitcoin transaction is not a ::Beacon Signal::.
+1. Set `hashBytes` to the 32 bytes in the `txOut`.
+1. If `signalSidecarData`:
+   1. Set `didUpdateBundle` to `signalSidecarData.updateBundle`
+   1. Set `bundleHashBytes` to the result of passing `didUpdateBundle` to the 
+      [JSON Canonicalization and Hash] algorithm.
+   1. If `bundleHashBytes` does not equal `hashBytes`, MUST raise an `invalidSidecarData` error. 
+      May identify Beacon Signal to resolver and request additional ::Sidecar data:: be provided.
```

Comment:
This "May" (and lots of other Mays) in this PR should be all caps?

Reply:
I don't have a good process for determining where the MUST MAY etc language should be used.

Also, probably in a separate issue we should discuss how exactly implementations can identify missing Beacon Signals data to resolvers.